producer.go

```diff
@@ -112,10 +125,44 @@ func (p *Producer) Produce(message []byte, opts ...ProduceOpt) error {
 
 }
 
+func (hdr *UserHeaders) validateHeaderKey(key string) error {
+	if strings.HasPrefix(key, "$memphis") {
+		return errors.New("Keys in headers should not start with $memphis")
+	}
+	return nil
+}
+
+func (hdr *UserHeaders) Add(key, value string) error {
+
+	err := hdr.validateHeaderKey(key)
+	if err != nil {
+		return err
+	}
+
+	header := Header{Key: key, Value: value}
+	hdr.Headers = append(hdr.Headers, header)
+	return nil
+}
+
 // ProducerOpts.produce - produces a message into a station using a configuration struct.
 func (opts *ProduceOpts) produce(p *Producer) error {
+	var hdrs []Header
+	hdrs = append(hdrs, Header{Key: "$memphis_connectionId", Value: p.conn.ConnId}, Header{Key: "$memphis_producedBy", Value: p.Name})
+
+	for _, userHdr := range opts.Headers.Headers {
+		hdrs = append(hdrs, userHdr)
+	}
+
+	headersMap := map[string][]string{}
```

Comment:
initialize this map with the memphis headers and after that iterate over the user headers and add them to this map, instead doing 2 for loops